### PR TITLE
if a value is wrapped, return the wrapped value from ractive.get - 2513

### PR DESCRIPTION
**Pull request by evs-chris**
**Description of the pull request:**
This adds a check for a wrapper in `Ractive.prototype.get`, and if it finds one, it returns the wrapped value whilst still respecting capture. This feels a little dirty, but it can't be done in `Model.get` because that would break the rest of the world. Should I add a new `get` method meant for external use to `Model` instead?

**Fixes the following issues:**
#2513 

**Is breaking:**
Quite the opposite.

**Reviewers:**
@martypdx, please

### CODE REVIEW

**Comment by skeptic35** [comment]
@evs-chris Firstly, you are now officially my hero :)
Secondly
>  Should I add a new get method meant for external use to Model instead?

If I understood that correctly, that wouldn't help in situations where I want to pass the Model on to a component like so:
```<my-component model=[[model]]/>```

**Comment by evs-chris** [comment]
@skeptic35, in this case, Model is actually an internal ractive class that represents a piece of data. Just an implementation detail and not anything that would affect API :smile:

**Comment by skeptic35** [comment]
Ok - I'll take your word on that ;)
Applied the patch to my Ractive source and it seems to work quite well. Now I can turn to the other errors I get ;)

**Comment by skeptic35** [comment]
Sorry, one more thing: Passing the model on to a function within a template. E. g.:
```{{doSomething(model)}}```
In 0.7.3 ```doSomething``` received the Backbone-Model, in 0.8 it receives the wrapped model (i. e. only the attributes). Also in 0.7.3 I could actually do something like this in my template:
```{{model.get("some_attribute")}}```
That also doesn't work anymore in 0.8
Same problem as the original one?

**Comment by evs-chris** [comment]
Ah, so expressions should get unwrapped values too. I haven't really used adaptors much, and it seems our test suite is a little lacking there. Out of curiosity though, why not `{{model.some_attribute}}`?

**Comment by martypdx** [comment]
@skeptic35, @evs-chris I think prior to 0.7 they used wrapped values, but I'm not 100% sure. It's an area where there's ambiguity in what is the "right thing" to do.

Consider these three uses: (see https://jsfiddle.net/martypdx/tbumr4st/):

```html
{{ model }} 
{{ model.toUpperCase() }}  //if adapted model was a string, you might want to do this
{{ method(model) }} //but maybe you want to pass the unadapted source to a method
```

**Comment by evs-chris** [comment]
I'm about to update this to have `Ractive.get` and expressions call `Model.get` and receive an unwrapped value. Non-expressions and chain-related `get`s will still get the adapted value. Does that sound appropriate?

**Comment by martypdx** [comment]
@evs-chris I just about to push almost identical changes :)

File: src/Ractive/prototype/get.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { splitKeypath } from '../../shared/keypaths';
 import resolveReference from '../../view/resolvers/resolveReference';
+import { capture } from '../../global/capture';
```
Comment (martypdx) [inline src/Ractive/prototype/get.js]: this can be removed now

**Comment by martypdx** [comment]
@evs-chris it doesn't appear in the current code there's a time we don't do `true, true`. Did I miss a usage? If that's true, we could just have a single param that means both things...

**Comment by evs-chris** [comment]
That's a fair point... if it's being captured, it's external/user-facing and should thus be unwrapped. Fixed!

**Comment by skeptic35** [comment]
@evs-chris 
> Out of curiosity though, why not ```{{model.some_attribute}}?```

```{{model.get("some_attribute")}}``` was simply a bad example. Actual point is to be able to call methods on the model. But I agree with @martypdx: There is amibguity as to what is the right thing to do. I could well live without being able to call methods of the wrapped model in the template, As long as ```{{doSomething(model)``` receives the unwrapped model that is ;)

**Pull request merged**